src/javascript/app/pages/user/account/settings/personal_details.js

```diff
@@ -402,17 +417,19 @@ const PersonalDetails = (() => {
                     }));
                 });
                 if (residence) {
-                    const $tax_residence = $('#tax_residence');
-                    $tax_residence.html($options_with_disabled.html()).promise().done(() => {
-                        setTimeout(() => {
-                            const residence_value = get_settings_data.tax_residence ?
-                                get_settings_data.tax_residence.split(',') : Client.get('residence') || '';
-                            $tax_residence.select2()
-                                .val(residence_value)
-                                .trigger('change')
-                                .setVisibility(1);
-                        }, 500);
-                    });
+                    if (shouldShowTax(get_settings_data)) {
+                        $tax_residence = $('#tax_residence');
+                        $tax_residence.html($options_with_disabled.html()).promise().done(() => {
+                            setTimeout(() => {
+                                const residence_value = get_settings_data.tax_residence ?
+                                    get_settings_data.tax_residence.split(',') : residence || '';
+                                $tax_residence.select2()
+                                    .val(residence_value)
+                                    .trigger('change')
+                                    .setVisibility(1);
+                            }, 500);
```

Comment:
may i know why do we use timeout here?

Reply:
I'm just changing the indentation here, but it was added originally because we need to wait for some scripts to load and we tried various ways of waiting but in the end this was the only thing that worked at the time